src/form.js

```diff
@@ -983,6 +983,21 @@ define([
             var _this = this;
             this._logicManager.forEachBrokenReference(updateReferences);
         },
+        isReferencedByOtherMugs: function (mug, except) {
```

Comment:
I meant I think it would make more sense if the workload for this was in the logic manager. This only uses the ufid of the mugs, and I usually think of the logic manager as doing everything with references